src/main/java/uk/gov/hmcts/reform/civil/service/PaymentsService.java

```diff
@@ -1,73 +1,112 @@
         var organisationName = organisationService.findOrganisationById(organisationId)
             .map(Organisation::getName)
             .orElseThrow(RuntimeException::new);
-
-        String customerReference = ofNullable(caseData.getClaimIssuedPaymentDetails())
-            .map(PaymentDetails::getCustomerReference)
-            .orElse(caseData.getPaymentReference());
-        CreditAccountPaymentRequest creditAccountPaymentRequest = null;
-
+        PBAServiceRequestDTO pbaServiceRequestDTO = null;
         if (!isSpecCaseCategory(caseData, featureToggleService.isAccessProfilesEnabled()))  {
-            creditAccountPaymentRequest = CreditAccountPaymentRequest.builder()
-                .accountNumber(caseData.getApplicantSolicitor1PbaAccounts().getValue().getLabel())
+            pbaServiceRequestDTO = PBAServiceRequestDTO.builder()
+                .accountNumber(hearingFeePBADetails.getApplicantsPbaAccounts()
+                                   .getValue().getLabel())
                 .amount(claimFee.getCalculatedAmount())
-                .caseReference(caseData.getLegacyCaseReference())
-                .ccdCaseNumber(caseData.getCcdCaseReference().toString())
-                .customerReference(customerReference)
-                .description("Claim issue payment")
+                .customerReference(hearingFeePBADetails.getPbaReference())
                 .organisationName(organisationName)
-                .service(paymentsConfiguration.getService())
-                .siteId(paymentsConfiguration.getSiteId())
-                .fees(new FeeDto[]{claimFee})
+                .idempotencyKey(String.valueOf(UUID.randomUUID()))
                 .build();
         } else if (isSpecCaseCategory(caseData, featureToggleService.isAccessProfilesEnabled())) {
-            creditAccountPaymentRequest = CreditAccountPaymentRequest.builder()
-                .accountNumber(caseData.getApplicantSolicitor1PbaAccounts().getValue().getLabel())
+            pbaServiceRequestDTO = PBAServiceRequestDTO.builder()
+                .accountNumber(hearingFeePBADetails.getApplicantsPbaAccounts()
+                                   .getValue().getLabel())
                 .amount(claimFee.getCalculatedAmount())
-                .caseReference(caseData.getLegacyCaseReference())
-                .ccdCaseNumber(caseData.getCcdCaseReference().toString())
-                .customerReference(customerReference)
-                .description("Claim issue payment")
+                .customerReference(hearingFeePBADetails.getPbaReference())
                 .organisationName(organisationName)
-                .service(paymentsConfiguration.getSpecService())
-                .siteId(paymentsConfiguration.getSpecSiteId())
-                .fees(new FeeDto[]{claimFee})
+                .idempotencyKey(String.valueOf(UUID.randomUUID()))
                 .build();
         }
```

Comment:
The content of the `if` and the `else if` is completely identical.

If it's meant to be, you don't need the `if` and `else if`. If it's not meant to be, then you have a problem.

Please take a look and either remove the unnecessary code or fix the issue.

Reply:
Also: install the SonarLint plugin on your IntelliJ, as it will detect these situations automatically for you.